### DISCUSSION TITLE
Kafka connector tasks are assigned on wrong Kafka connect cluster

### DISCUSSION THREAD

**Discussion by ebuildy** [strimzi/strimzi-kafka-operator#10063] [Q&A] [answered]
Have 2 connectors: debezium and snowflake I want to separate on different kafka connect clusters.

So I use KafkaConnector labels `strimzi.io/cluster`:

<img width="932" alt="Capture d’écran, le 2024-05-05 à 20 18 05" src="https://github.com/strimzi/strimzi-kafka-operator/assets/1219817/420b208b-cb11-4aa7-a37d-4d6d7fa82212">

In the other side, I declare 2 KafkaConnect clusters:

<img width="512" alt="Capture d’écran, le 2024-05-05 à 20 18 36" src="https://github.com/strimzi/strimzi-kafka-operator/assets/1219817/a1c04d5d-d194-495f-badd-e358bfacde1b">

`strimzi.io/use-connector-resources=true` annotation is set.

But I notice snowflake tasks are running on debezium KafkaConnect cluster!  It looks like the 9 snowflake tasks are running twice on both cluster:

<img width="1788" alt="Capture d’écran, le 2024-05-05 à 20 25 43" src="https://github.com/strimzi/strimzi-kafka-operator/assets/1219817/b29a9881-df2c-42f5-981c-1a098d84b7c3">

**Comment by scholzj** [comment, marked as answer]
Did you followed the docs covering how to deploy multiple Connect clusters connected to the same Kafka cluster?

**Reply by ebuildy** [reply]
ho didnt configure topics well :-/ Thanks you so much, it's working fine now.

Why the operator is not taking care of that?

**Reply by scholzj** [reply]
> Why the operator is not taking care of that?

Sins of the past. I'm afraid we made the topic names and group.id years ago as optional configs with static default values. Only later we realized that issues like this will be more common that expected. But it cannot be changed easily now as it would break things for many existing deployments. 😭

**Reply by ebuildy** [reply]
ho I understand, this is classic :-) maybe a feature-flag could help
